tests/unittests/OnnxExporterTest.cpp

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "glow/ExecutionEngine/ExecutionEngine.h"
+#include "glow/Exporter/ONNXModelWriter.h"
+#include "glow/Graph/Graph.h"
+#include "glow/Graph/Nodes.h"
+#include "glow/Graph/PlaceholderBindings.h"
+#include "glow/Importer/ONNXModelLoader.h"
+#include "gtest/gtest.h"
+
+#include "llvm/Support/FileSystem.h"
+
+#ifndef GLOW_DATA_PATH
+#define GLOW_DATA_PATH
+#endif
+
+using namespace glow;
+
+namespace {
+void testLoadAndSaveONNXModel(const std::string &name) {
+  ExecutionEngine EE{};
+  auto &mod = EE.getModule();
+  Function *F = mod.createFunction("main");
+
+  llvm::Error err = llvm::Error::success();
+  ONNXModelLoader onnxLD(name, {}, {}, *F, &err);
+
+  if (err) {
+    llvm::errs() << "ONNXModelLoader failed to load model: " << name << ".\n";
+    return;
+  }
+  std::string outputFilename(name + ".output.onnxtxt");
+  { ONNXModelWriter onnxWR(outputFilename, *F, 1, 1, &err, true); }
+  llvm::sys::fs::remove(outputFilename);
+  EXPECT_FALSE(err) << "file name: " << name;
+}
+} // namespace
+
+TEST(exporter, onnxModels) {
+  std::string inputDirectory(GLOW_DATA_PATH "tests/models/onnxModels");
+  std::error_code code;
+  llvm::sys::fs::directory_iterator dirIt(inputDirectory, code);
+  for (llvm::sys::fs::directory_iterator dirIt(inputDirectory, code);
+       !code && dirIt != llvm::sys::fs::directory_iterator();
+       dirIt.increment(code)) {
+    auto name = dirIt->path();
+    if (name == "tests/models/onnxModels/preluInvalidBroadcastSlope.onnxtxt") {
+      continue;
+    }
+    testLoadAndSaveONNXModel(dirIt->path());
```

Comment:
Would it be possible to do something like load a proto, export it, load it back in, and run it? You could check the results are the same as if you just load/run it.

Reply:
That was my plan for the next PR, because I would need to modify ONNXModelLoader to support Glow specific nodes first.